src/bindings/python/src/pyopenvino/core/infer_request.cpp

```diff
@@ -612,6 +612,19 @@ void regclass_InferRequest(py::module m) {
             :rtype: List[openvino.runtime.VariableState]
         )");
 
+    cls.def(
+        "reset_state",
+        [](InferRequestWrapper& self) {
+            return self.m_request.reset_state();
+        },
+        py::call_guard<py::gil_scoped_release>(),
```

Comment:
Not sure if GIL should be released here. As it is "void" function, it could overlap with another `infer` call in the next line.
```suggestion
```

Reply:
done